namespace/namespace.go

```diff
@@ -12,19 +12,48 @@ type namespaceStore struct {
 	store     corekv.Store
 }
 
-var _ corekv.Store = (*namespaceStore)(nil)
+var _ corekv.TxnStore = (*namespaceStore)(nil)
+
+type namespacedTxn struct {
+	corekv.Store
+
+	txn corekv.Txn
+}
+
+var _ corekv.Txn = (*namespacedTxn)(nil)
 
 // Wrap lets you namespace a store with a given prefix.
-// The generic parameter allows you to wrap any kind of
-// underlying store ( corekv.TxnStore, Batchable, etc) and keep
-// the concrete type, just wrapped.
 func Wrap(store corekv.Store, prefix []byte) corekv.Store {
 	return &namespaceStore{
 		namespace: prefix,
 		store:     store,
 	}
 }
 
+// WrapTS lets you namespace a transaction store with a given prefix.
+func WrapTS(store corekv.TxnStore, prefix []byte) corekv.TxnStore {
+	return &namespaceStore{
+		namespace: prefix,
+		store:     store,
+	}
+}
+
+// WrapTxn lets you namespace a transaction with a given prefix.
+func WrapTxn(txn corekv.Txn, prefix []byte) corekv.Txn {
+	return &namespacedTxn{
+		Store: Wrap(txn, prefix),
+		txn:   txn,
+	}
+}
+
+func (nstore *namespaceStore) NewTxn(readonly bool) corekv.Txn {
```

Comment:
thought: I think returning interfaces is bad practice as I've said many times before (Unless we want to return multiple different concrete types which is not the case here). We can just make the concrete type public and keep the fields private. A lot less limitations and better compiler optimization as a result.